include/swift-c/SyntaxParser/SwiftSyntaxParser.h

```diff
@@ -232,6 +232,65 @@ swiftparse_parse_string(swiftparse_parser_t, const char *source);
 /// declarations, etc.
 SWIFTPARSE_PUBLIC const char* swiftparse_syntax_structure_versioning_identifier(void);
 
+typedef struct {
+  /// Represents the range for the fixit.
+  swiftparse_range_t range;
+  /// Represent the text for replacement.
+  const char* text;
+} swiftparse_fixit_t;
+
+enum swiftparser_diagnostic_severity {
+  swiftparser_diagnostic_severity_Error = 0,
+  swiftparser_diagnostic_severity_Warning = 1,
+  swiftparser_diagnostic_severity_Remark = 2,
+  swiftparser_diagnostic_severity_Note = 3,
+};
+
+/// This is for the client to provide an opaque pointer that the parser will
+/// associate with a diagnostic.
+/// This pointer is only valid when received from the argument of a
```

Comment:
Bit better to re-phrase "this pointer is only valid to access from within the swiftparse_diagnostic_handler_t block